models/userStatus.js

```diff
@@ -52,92 +51,59 @@ const getGroupRole = async (rolename) => {
+    const groupRoleObj = await getGroupRole("group-idle");
+    if(groupRoleObj?.roleExists) {
+      const groupIdleRoleId = groupRoleObj.role.roleid;
+      const user = await usersCollection.doc(userId).get();
+      const discordId = user.data().discordId;
+      if(discordId) {
+        // Remove role from firestore collection
+        const hasRole = await memberRoleModel
+          .where("roleid", "==", groupIdleRoleId)
+          .where("userid", "==", discordId)
+          .limit(1)
+          .get();
+        if (!hasRole.empty) {
+          const oldRole = [];
+          hasRole.forEach((role) => oldRole.push({ id: role.id }));
+          await memberRoleModel.doc(oldRole[0].id).delete();
+        }
 
-      // remove role from member role collection in firestore
-      const hasRole = await memberRoleModel
-        .where("roleid", "==", ROLE_ID)
-        .where("userid", "==", USER_DISCORD_ID)
-        .limit(1)
-        .get();
-      if (!hasRole.empty) {
-        const oldRole = [];
-        hasRole.forEach((role) => oldRole.push({ id: role.id }));
-        await memberRoleModel.doc(oldRole[0].id).delete();
+        await removeRoleFromUser(groupIdleRoleId, discordId);
       }
-
-      const dataForDiscord = {
-        roleid: ROLE_ID,
-        userid: USER_DISCORD_ID,
-      };
-      const authToken = jwt.sign({}, config.get("rdsServerlessBot.rdsServerLessPrivateKey"), {
-        algorithm: "RS256",
-        expiresIn: config.get("rdsServerlessBot.ttl"),
-      });
-
-      await fetch(`${DISCORD_BASE_URL}/roles`, {
-        method: "DELETE",
-        body: JSON.stringify(dataForDiscord),
-        headers: { "Content-Type": "application/json", Authorization: `Bearer ${authToken}` },
-      }).then((response) => response.json());
     }
   } catch (error) {
-    logger.error(`error in removing role from discord user. Reason - ${error}`);
+    logger.error(`error in removing group-idle role from discord user. Reason - ${error}`);
     throw error;
   }
 };
 
-const addGroupRoleToDiscordUser = async ({ userId, roleName, roleId, discordId }) => {
+const addGroupIdleRoleToDiscordUser = async (userId) => {
   try {
-    let groupRole;
-    if (!roleId) groupRole = await getGroupRole(roleName);
-    if (groupRole?.roleExists || roleId) {
-      const ROLE_ID = roleId || groupRole.role.roleid;
-      let userData;
-      if (!discordId) {
-        const user = await usersCollection.doc(userId).get();
-        userData = user.data();
-      }
-      const USER_DISCORD_ID = discordId || userData.discordId;
+    const groupRoleObj = await getGroupRole("group-idle");
+    if(groupRoleObj?.roleExists) {
+      const groupIdleRoleId = groupRoleObj.role.roleid;
+      const user = await usersCollection.doc(userId).get();
+      const discordId = user.data().discordId;
+      if(discordId) {
+        // Add role to firestore collection
+        const alreadyHasRole = await memberRoleModel
```

Comment:
if the user already has the groupIdleRoleId role, but the where clause is missing the array_contains operator. The correct code should be:

const alreadyHasRole = await memberRoleModel
  .where("roleid", "==", groupIdleRoleId)
  .where("userid", "==", discordId)
  .where("array_contains", "roles", groupIdleRoleId)
  .limit(1)
  .get();

Reply:
The member-discord-roles collection has a document that has a single role like
{
  roleid: 122333,
  userid: 2321312
}

Document is for single role only there is no array in it.